internal/gcs/process.go

```diff
@@ -220,6 +283,12 @@ func (p *Process) ResizeConsole(ctx context.Context, width, height uint16) (err
 		trace.StringAttribute("cid", p.cid),
 		trace.Int64Attribute("pid", int64(p.id)))
 
+	p.gcLock.RLock()
+	defer p.gcLock.RUnlock()
+	if p.gc == nil {
```

Comment:
Was this before the introduction of our friend p.gcClosed()?

Reply:
`gcClosed` `RLocks`, and I dont think `RWMutex` is re-entrant (The docs say "It should not be used for recursive read locking"), so I would need to rewrite it as:
``` go
if p.gcClosed() { 		
	return hcs.ErrAlreadyClosed
}
p.gcLock.RLock()
defer p.gcLock.RUnlock()
// ...
```

But that allows a race condition where another process closes `gc` inbetween the check and the rpc call

Reply:
As a side note, I considered:
```go
func (p *Process) gcClosed() bool {
	p.gcLock.RLock()
	defer p.gcLock.RUnlock()
	return gcclosed()	
}

// unsafe: requires holding p.gcLock
func (p *Process) gcclosed() bool { return p.gc == nil }
```

but that seems .... excessive